jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -682,6 +666,153 @@ ecma_string_to_utf8_string (const ecma_string_t *string_desc_p, /**< ecma-string
+
+#define ECMA_STRING_IS_ASCII(char_p, size) ((size) == lit_utf8_string_length ((char_p), (size)))
+
+/**
+ * Return the raw byte array of the string, if it is available.
+ *
+ * @return byte array start - if the byte array of a string is available
+ *         NULL - otherwise
+ */
+const lit_utf8_byte_t *
+ecma_string_raw_chars (const ecma_string_t *string_p, /* ecma-string */
+                       lit_utf8_size_t *size_p, /* size of the ecma string */
+                       bool *is_ascii_p) /* true, if ascii string (size == length) */
+{
+  ecma_length_t length;
+  lit_utf8_size_t size;
+  const lit_utf8_byte_t *result_p;
+
+  switch (ECMA_STRING_GET_CONTAINER (string_p))
+  {
+    case ECMA_STRING_CONTAINER_LIT_TABLE:
+    {
+      lit_literal_t lit = lit_get_literal_by_cp (string_p->u.lit_cp);
+      JERRY_ASSERT (LIT_RECORD_IS_CHARSET (lit));
+
+      length = lit_charset_literal_get_length (lit);
+      size = lit_charset_literal_get_size (lit);
+      result_p = lit_charset_literal_get_charset (lit);
+      break;
+    }
+    case ECMA_STRING_CONTAINER_HEAP_CHUNKS:
+    {
+      const ecma_string_heap_header_t *data_p = ECMA_GET_NON_NULL_POINTER (ecma_string_heap_header_t,
+                                                                           string_p->u.collection_cp);
+      size = data_p->size;
+      length = data_p->length;
+      result_p = (const lit_utf8_byte_t *) (data_p + 1);
+      break;
+    }
+    case ECMA_STRING_CONTAINER_HEAP_NUMBER:
+    {
+      size = ecma_string_get_heap_number_size (string_p->u.number_cp);
+
+      /* All numbers must be ascii strings. */
+      JERRY_ASSERT (ecma_string_get_length (string_p) == size);
+
+      length = size;
+      result_p = NULL;
+      break;
+    }
+    case ECMA_STRING_CONTAINER_UINT32_IN_DESC:
+    {
+      size = (lit_utf8_size_t) ecma_string_get_number_in_desc_size (string_p->u.uint32_number);
+
+      /* All numbers must be ascii strings. */
+      JERRY_ASSERT (ecma_string_get_length (string_p) == size);
+
+      length = size;
+      result_p = NULL;
+      break;
+    }
+    case ECMA_STRING_CONTAINER_MAGIC_STRING:
+    {
+      size = lit_get_magic_string_size (string_p->u.magic_string_id);
+
+      length = size;
+      result_p = lit_get_magic_string_utf8 (string_p->u.magic_string_id);
+
+      /* All magic strings must be ascii strings. */
+      JERRY_ASSERT (ECMA_STRING_IS_ASCII (result_p, size));
+      break;
+    }
+    default:
+    {
+      JERRY_ASSERT (ECMA_STRING_GET_CONTAINER (string_p) == ECMA_STRING_CONTAINER_MAGIC_STRING_EX);
+      size = lit_get_magic_string_ex_size (string_p->u.magic_string_ex_id);
+
+      length = size;
+      result_p = lit_get_magic_string_ex_utf8 (string_p->u.magic_string_ex_id);
+
+      /* All extended magic strings must be ascii strings. */
+      JERRY_ASSERT (ECMA_STRING_IS_ASCII (result_p, size));
+      break;
+    }
+  }
+
+  *size_p = size;
+  *is_ascii_p = (length == size);
```

Comment:
I don't think we gain too much. I was thinking about that myself but choose this cleaner solution at the end.